lib/evm/instruction/impl.ex

```diff
@@ -50,240 +51,208 @@ defmodule EVM.Instruction.Impl do
 
   ## Examples
 
-      iex> EVM.Instruction.Impl.add([1, 2], %{stack: []})
-      %{stack: [3]}
+      iex> EVM.Instruction.Impl.add([1, 2], %{})
+      3
```

Comment:
Can you add to the typespec for `op_result` the possibility of returning `EVM.val`? The behavior (if I understand) is that any impl that returns a numeric has that value pushed onto the stack? We should make sure that we actually expect an `EVM.val` and not a `numeric()`, however, so that dialyzer catches any numeric we return that we haven't `wrap_int`'d and `encode_signed`'d.

Reply:
Ah, yep!

Still getting used to typespecs...

[Allow op_result to be a EVM.val type](https://github.com/exthereum/evm/pull/2/commits/433d43acb5722265e53fb73980143cbaa43057fa)